### PR TITLE
Prevent compile errors on OSX

### PR DESCRIPTION
**Pull request by ClausKlein**
clangtidy cmake files too
fix possible buffer overflow too

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -23,50 +21,57 @@ foreach(mock_func IN LISTS ${test}_mock_funcs)
...
 
+########################################
+## TODO: what kind of hack is this? CK #
 foreach(src IN LISTS libsrc)
     list(APPEND test_srcs "../${src}")
 endforeach()
 add_library(testobj OBJECT ${test_srcs})
+########################################
+
+#TBD target_link_libraries(testobj netconf2)
```
Comment (michalvasko) [inline tests/CMakeLists.txt]: `testobj` does not need to be linked with `libnetconf2`, it is `libnetconf2`.

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -23,50 +21,57 @@ foreach(mock_func IN LISTS ${test}_mock_funcs)
...
     endif()
 
-    if (ENABLE_TLS)
+    if(ENABLE_TLS)
         list(APPEND client_tests test_client_tls)
     endif()
 endif()
 
+########################################
+## TODO: what kind of hack is this? CK #
```
Comment (michalvasko) [inline tests/CMakeLists.txt]: Not sure what the problem here is.

File: CMakeLists.txt [hunk 3]
```diff
@@ -196,22 +213,28 @@ check_function_exists(pthread_rwlockattr_setkind_np HAVE_PTHREAD_RWLOCKATTR_SETK
...
     endif()
+
+    #TODO target_link_libraries(netconf2 PUBLIC OpenSSL::SSL OpenSSL::Crypto)
     target_link_libraries(netconf2 ${OPENSSL_LIBRARIES})
     include_directories(${OPENSSL_INCLUDE_DIR})
 endif()
 
 # dependencies - libssh
-if(ENABLE_SSH)
+if(ENABLE_SSH AND OPENSSL_FOUND)
```
Comment (michalvasko) [inline CMakeLists.txt]: SSH does not use OpenSSL directly to my knowledge, so if you want to have only SSH support, you do not need it. Should also probably be fixed in the `if` above.

**Comment by ClausKlein** [comment]
Next week I’ll fix it

Regards,
Claus


Am Montag, 13. Januar 2020 schrieb Michal Vasko <notifications@github.com>:

**Comment by michalvasko** [comment]
Hi,
okay, seems fine now, thanks.

Regards,
Michal

**Pull request merged**